### PR TITLE
Fix/comment edit

### PR DESCRIPTION
**Pull request by ghost**
## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/hooks/src/use-comments.new.ts [hunk 1]
```diff
@@ -170,12 +170,13 @@ export function useCreateComment() {
...
 
 // hook used for editing a comment
-export function useEditComment(commentID: string) {
+export function useEditComment(commentID: string, isComment: boolean) {
   const sdk = getSDK();
   const queryClient = useQueryClient();
 
   return useMutation(
     (comment: IPublishData & { postID: string }) => {
+      if (!isComment) return;
```
Comment (kenshyx) [inline ui/hooks/src/use-comments.new.ts]: The use case for this flag is to prevent posting empty data?

Comment (ghost) [reply ui/hooks/src/use-comments.new.ts]: Yes, throws errors in the console otherwise

Comment (kenshyx) [reply ui/hooks/src/use-comments.new.ts]: I would suggest to rename this into something more verbose for the use case, like `hasCommentData`

File: ui/hooks/src/use-comments.new.ts [hunk 2]
```diff
@@ -170,12 +170,13 @@ export function useCreateComment() {
 }
 
 // hook used for editing a comment
-export function useEditComment(commentID: string) {
+export function useEditComment(commentID: string, isComment: boolean) {
   const sdk = getSDK();
   const queryClient = useQueryClient();
 
   return useMutation(
     (comment: IPublishData & { postID: string }) => {
```
Comment (kenshyx) [inline ui/hooks/src/use-comments.new.ts]: this function should be `async`

File: ui/hooks/src/use-comments.new.ts (hunk 1 above)
Comment (kenshyx) [reply ui/hooks/src/use-comments.new.ts]: the return should be consistent, `return Promise.resolve()`

File: ui/hooks/src/use-comments.new.ts (hunk 2 above)
Comment (kenshyx) [reply ui/hooks/src/use-comments.new.ts]: the mutation function, not the hook :D

Comment (kenshyx) [reply ui/hooks/src/use-comments.new.ts]:
```
return useMutation(
    async (comment: IPublishData & { postID: string }) => {
 ```

File: ui/hooks/src/use-comments.new.ts [hunk 3]
```diff
@@ -170,13 +170,13 @@ export function useCreateComment() {
...
 // hook used for editing a comment
-export function useEditComment(commentID: string, hasCommentData: boolean) {
+export async function useEditComment(commentID: string, hasCommentData: boolean) {
   const sdk = getSDK();
   const queryClient = useQueryClient();
 
   return useMutation(
     (comment: IPublishData & { postID: string }) => {
-      if (!hasCommentData) return;
+      if (!hasCommentData) return Promise.resolve(null);
```
Comment (kenshyx) [inline ui/hooks/src/use-comments.new.ts]: You can just use Promise.resolve() which will resolve to `undefined` value like `return;`

**Review by kenshyx — APPROVED** [review]

**Pull request merged**